src/workerd/api/sockets.h

```diff
@@ -139,8 +150,13 @@ class Socket: public jsg::Object {
     JSG_READONLY_PROTOTYPE_PROPERTY(closed, getClosed);
     JSG_READONLY_PROTOTYPE_PROPERTY(opened, getOpened);
     JSG_READONLY_PROTOTYPE_PROPERTY(upgraded, getUpgraded);
+    JSG_READONLY_PROTOTYPE_PROPERTY(secureTransport, getSecureTransport);
     JSG_METHOD(close);
     JSG_METHOD(startTls);
+
+    JSG_TS_OVERRIDE({
+      get secureTransport(): 'on' | 'off' | 'starttls';
```

Comment:
```suggestion
      readonly secureTransport: 'on' | 'off' | 'starttls';
```

Reply:
This follows the same pattern of other methods defined by `JSG_READONLY_PROTOTYPE_PROPERTY`. I think we should do this on the type generator level. If not, there will be inconsistency.

Reply:
ok, that's fine.